osu.Game/Rulesets/Objects/HitObject.cs

```diff
@@ -35,6 +35,11 @@ public class HitObject
         /// </summary>
         public event Action<HitObject> DefaultsApplied;
 
+        /// <summary>
+        /// Invoked before <see cref="NestedHitObjects"/> are cleared.
+        /// </summary>
+        internal event Action<HitObject> RemoveNestedHitObjects;
```

Comment:
I'm not a huge fan of this event either. Honestly wondering whether just subscribing to `DefaultsApplied` and cleaning up all of the old entries manually wouldn't be better.

Or, heck, making `NestedHitObjects` a `BindableList`, but that sounds like it might not be very pleasant at all...

Reply:
Yeah, I really don't like this either. There has to be a better way...